client/asset/dcr/externaltx.go

```diff
@@ -388,32 +366,30 @@ func (dcr *ExchangeWallet) isOutputSpent(ctx context.Context, output *outputSpen
 // If no tx is found to spend the provided output, the hash of the block that
 // was last checked is returned along with any error that may have occurred
 // during the search.
-func (dcr *ExchangeWallet) findTxOutSpender(ctx context.Context, op outPoint, outputPkScript []byte, startBlock *block) (*chainjson.TxRawResult, *chainhash.Hash, error) {
+func (dcr *ExchangeWallet) findTxOutSpender(ctx context.Context, op outPoint, outputPkScript []byte, startBlock *block) (*wire.MsgTx, *chainhash.Hash, int64, error) {
 	var lastScannedHash *chainhash.Hash
 
 	iHeight := startBlock.height
 	iHash := startBlock.hash
 	bestBlock := dcr.cachedBestBlock()
 	for {
-		blockFilter, err := dcr.getBlockFilterV2(ctx, iHash)
+		key, filter, err := dcr.wallet.BlockFilter(ctx, iHash)
 		if err != nil {
-			return nil, lastScannedHash, err
+			return nil, lastScannedHash, 0, err
```

Comment:
0 -> iHeight. There may have been a scan in a previous loop iteration, especially if `lastScannedHash != nil`.